radas/adas_interface/fortran_reader.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+from fortranformat import FortranRecordReader as FReader
+
+def read_single_values(list_of_lines, format_spec, start_at_line):
+    """
+    Makes a 'generator', which is a special Python object that returns a value each time that it is called.
+
+    This generator reads each line in the list of lines, and tries to format it according to format_spec
+    For instance, the default format '5e16.9' will try to read 5 fortran-formatted floats in exponential format
+    with a width of 16 characters and 9 characters after the decimal point
+    Where there are less than 5 numbers on a line, the fortran reader will return 'None'. These are skipped (not
```

Comment:
What happens if one specifies the wrong format? Will there not be any errors or warnings then?

Reply:
It makes a slightly cryptic error. I've extended this to
```
ValueError('fortranformat could not read line 5:    -.69877   -.52268   -.30083   -.15470    .00020    .17629    .30123    .47732 with format (8f8.5)\nMake sure that your format_spec has the correct number of elements, the correct spacing and the correct format for the line.\nSee https://docs.oracle.com/cd/E19957-01/805-4939/6j4m0vn9f/index.html for reference.\nfortranformat error was: 77   -.5 is not a valid input as for an E, ES, EN or D edit descriptor')
```